adapters/tagbidder/vast_tag_response_handler.go

```
@@ -79,8 +81,18 @@ func (handler *VASTTagResponseHandler) vastTagToBidderResponse(internalRequest *
 	}
 
 	typedBid := &adapters.TypedBid{
-		Bid:     &openrtb.Bid{},
-		BidType: openrtb_ext.BidTypeVideo,
+		Bid:      &openrtb.Bid{},
+		BidType:  openrtb_ext.BidTypeVideo,
+		BidVideo: &openrtb_ext.ExtBidPrebidVideo{},
+	}
+
+	creatives := adElement.FindElements("Creatives/Creative")
+	if nil != creatives {
+		for _, creative := range creatives {
+			// get duration. Ignore errors
+			dur, _ := getDuration(creative)
+			typedBid.BidVideo.Duration = int(dur) // prebid expects int value
+		}
```

1. Iterates over each creative provided in the VAST
2. Converting from `float64` to `int`. Because Prebid Server expects video duration in `int`
> if d = "0:0:30.5"  => 30 seconds and 5 milliseconds
   `getDuration` will return `30.005` seconds
   But, `typedBid.BidVideo.Duration` will be set to `30` seconds (0.005 seconds will be dropped)
3. Errors are ignored. If `getDuration` failed then `dur` will be `0.0`. Hence, `typedBid.BidVideo.Duration` will be `0`